cli/tests/pcluster/config/test_validators.py

```diff
@@ -545,6 +546,69 @@ def test_s3_validator(mocker, boto3_stubber, config, num_calls, bucket, expected
     utils.assert_param_validator(mocker, config, expected_message)
 
 
+@pytest.mark.parametrize(
+    "bucket, error_code, expected_message",
+    [
+        (
+            "bucket",
+            None,
+            "cluster_resource_bucket must be in the same region of the cluster.",
+        ),
+        (
+            "not_existed_bucket",
+            "NoSuchBucket",
+            "The S3 bucket 'not_existed_bucket' does not appear to exist",
+        ),
+        (
+            "access_denied_bucket",
+            "AccessDenied",
+            "You do not have access to the S3 bucket 'access_denied_bucket'",
+        ),
+    ],
+)
+def test_s3_bucket_region_validator(mocker, boto3_stubber, error_code, bucket, expected_message):
+    os.environ["AWS_DEFAULT_REGION"] = "eu-west-1"
+    get_bucket_location_response = {
+        "ResponseMetadata": {},
+        "LocationConstraint": "af-south1",
```

Comment:
`us-east-1` is an edge case, so we should add a test case for that.
We also need to add a test case where the validation passes for full coverage.

Reply:
Added. Also added explainations in the comments of the code